### DISCUSSION TITLE
Make getLoadContext async for all adapters

### DISCUSSION THREAD

**Discussion by arielvieira** [remix-run/remix#6167] [Proposals]
Right now the `getLoadContext` it's already async for [@remix-run/deno](https://www.npmjs.com/package/@remix-run/deno)
https://github.com/remix-run/remix/blob/935464d663f2c993082fe46615f509243464c0a9/packages/remix-deno/server.ts#L29

But is not async for the other adapters, for now I'm patching the cloudflare adapter so I can use it as async, but would be really awesome to have it async by default for every other providers beside deno

#### `getLoadContext` code from other adapters
**@remix-run/architect**
https://github.com/remix-run/remix/blob/935464d663f2c993082fe46615f509243464c0a9/packages/remix-architect/server.ts#L53
**@remix-run/cloudflare-pages**
https://github.com/remix-run/remix/blob/935464d663f2c993082fe46615f509243464c0a9/packages/remix-cloudflare-pages/worker.ts#L31
**@remix-run/cloudflare-workers**
https://github.com/remix-run/remix/blob/935464d663f2c993082fe46615f509243464c0a9/packages/remix-cloudflare-workers/worker.ts#L40
**@remix-run/express**
https://github.com/remix-run/remix/blob/935464d663f2c993082fe46615f509243464c0a9/packages/remix-express/server.ts#L56
**@remix-run/netlify**
https://github.com/remix-run/remix/blob/935464d663f2c993082fe46615f509243464c0a9/packages/remix-netlify/server.ts#L50
**@remix-run/vercel**
https://github.com/remix-run/remix/blob/935464d663f2c993082fe46615f509243464c0a9/packages/remix-vercel/server.ts#L50

And even in hydrogen it is async as well
https://github.com/Shopify/hydrogen/blob/2023-04/packages/remix-oxygen/src/server.ts#L21

**Comment by MichaelDeBoey** [comment]
Created a PR to support this out of the box: https://github.com/remix-run/remix/pull/6170

**Reply by kiliman** [reply]
Nice. Just note that Express handlers are *not* async by default, so you'll need something like `express-async-handler` or `@awaitjs/express`

https://zellwk.com/blog/async-await-express/

```ts
import asyncHandler from 'express-async-handler'

app.all('*', asyncHandler(handleRequest))

async function handleRequest(
  req: express.Request,
  res: express.Response,
  next: express.NextFunction,
) {
  if (MODE === 'development') {
    purgeRequireCache()
  }
  let build = require(BUILD_DIR)
  let loadContext = await getLoadContext(req, res)
  
  return createRequestHandler({
    build,
    mode: MODE,
    getLoadContext: loadContext
  })(req, res, next)
}
```

**Reply by MichaelDeBoey** [reply]
`requestHandler` was already async typed though
https://github.com/remix-run/remix/blob/73e7517ccfc7c524878dd0d3762bd21899630a91/packages/remix-express/server.ts#L29-L33

**Reply by kiliman** [reply]
Ok, perhaps this was what I was referencing

https://github.com/remix-run/remix/blob/73e7517ccfc7c524878dd0d3762bd21899630a91/packages/remix-express/server.ts#L64-L68

**Reply by arielvieira** [reply]
That was fast @MichaelDeBoey, thanks! 😄

**Reply by MichaelDeBoey** [reply]
@arielvieira I thought all of them where already async tbh 🤔🙈
